src/Entity/Goal.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace Acquia\LiftClient\Entity;
+
+use Acquia\LiftClient\Exception\LiftSdkException;
+use Acquia\LiftClient\Utility\Utility;
+use Symfony\Component\Serializer\Serializer;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\Normalizer\CustomNormalizer;
+
+class Goal extends \ArrayObject
```

Comment:
Right now, class Goal/Slot/Visibility extends \ArrayObject, and uses EntityValueTrait/EntityTrait to share common functions. Is it possible that we define an abstract Acquia\LiftClient\Entity\Entity that extends \ArrayObject, and then having it encapsulate all the shared functions and logic? In this way, class Goal/Slot/Visibility can all just extend Entity.